### PR TITLE
Fix various issues

### PR DESCRIPTION
**Pull request by akram**

### CODE REVIEW

**Comment by akram** [comment]
/assign @gabemontero 
/assign @adambkaplan 

Hi team,

these 2 commits fixes:
- a security issue present in Jackson and for which GitHub displays a warning on the repo landing page
- some warning about incomplete javadoc
Can you PTAL ?

File: pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
   </parent>
   <groupId>io.fabric8.jenkins.plugins</groupId>
   <artifactId>openshift-sync</artifactId>
-  <version>1.0.40-SNAPSHOT</version>
+  <version>1.0.41-SNAPSHOT</version>
```
Comment (gabemontero) [inline pom.xml]: You should rebase this PR to pick up https://github.com/openshift/jenkins-sync-plugin/commit/1baa73027d14452da976f60050c78afeed3d6c80#diff-600376dffeb79835ede4a0b285078036 from https://github.com/openshift/jenkins-sync-plugin/pull/322 and not "re-introduce" this change with this PR.

**Comment by akram** [comment]
@gabemontero rebased

**Pull request merged**